lib/common/client/middleware/response/vetext_errors.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      module Response
+        class VETextError < Faraday::Response::Middleware
+          def on_complete(env)
+            return if env.success?
+
+            case env.status
+            when 400..499
+              raise Common::Exceptions::BackendServiceException.new('VETEXT_PUSH_400', {}, env.status,
+                                                                    env.body)
+            when 500..599
+              raise Common::Exceptions::BackendServiceException.new('VETEXT_PUSH_502', {}, env.status)
+            else
+              raise env
```

Comment:
Is this case covered by specs? I'm not sure what happens if you raise on even but I think it would be a StandardError. That is if it's even possible to get here 😄 Maybe have it be a generic `Common::Exceptions::BackendServiceException`

Reply:
It's covered in service_spec.rb. If you think it's more appropriate to have it just be a generic `Common::Exceptions::BackendServiceException` that's easy enough to change. I'm not exactly sure what the advantages/disadvantages of having it be either or would be.

Reply:
Looks like if you raise on `env` which is a `Faraday::Env` struct you'll get the following:
```
pry(#<Common::Client::Middleware::Response::VETextError>)> raise env
TypeError: exception class/object expected
```

Reply:
After refreshing myself on what the http status codes are supposed to mean, I do think 502 the most accurate choice for status code.